src/state/slices/color_ramp_slice.ts

```diff
@@ -135,6 +139,40 @@ export const addColorRampDerivedStateSubscribers = (store: SubscribableStore<Dat
     })
   );
 
+  // Update color ramp range if the threshold changes for the currently selected feature
+  addDerivedStateSubscriber(
+    store,
+    (state) => [state.thresholds],
+    ([thresholds], [prevThresholds]) => {
+      const dataset = store.getState().dataset;
+      const featureKey = store.getState().featureKey;
+      if (dataset === null || featureKey === null) {
+        return;
+      }
+      const featureData = dataset.getFeatureData(featureKey);
+      if (!featureData) {
+        throw new Error(
+          `ViewerStateStore: Expected feature data not found for key '${featureKey}' in Dataset when updating color ramp range from threshold.`
+        );
+      }
+      // Check if the threshold on the currently selected feature has changed. If so, reset the color ramp range to match
+      // the new threshold.
+      const prevThreshold = prevThresholds.find(thresholdMatchFinder(featureKey, featureData.unit));
+      const newThreshold = thresholds.find(thresholdMatchFinder(featureKey, featureData.unit));
+      if (newThreshold && isThresholdNumeric(newThreshold)) {
+        if (
+          !prevThreshold ||
+          !isThresholdNumeric(prevThreshold) ||
+          newThreshold.min !== prevThreshold.min ||
+          newThreshold.max !== prevThreshold.max
+        ) {
+          return { colorRampRange: [newThreshold.min, newThreshold.max] as [number, number] };
+        }
+      }
+      return;
+    }
```

Comment:
might explicitly `return undefined` here if TS is gonna force you to put a `return` statement at the end of this function